src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx

```diff
@@ -175,6 +176,23 @@ export default class IntegrationRepos extends AsyncComponent {
     );
   }
 
+  renderError(error) {
+    const badRequest = Object.values(this.state.errors).find(
+      resp => resp && resp.status === 400
+    );
+    if (badRequest) {
+      return (
+        <Alert type="error" icon="icon-circle-exclamation">
+          {t(
+            'We were unable to fetch repositories for this integration. Try again later, or reconnect this integration.'
```

Comment:
@markstory Why would trying again later resolve a bad request error? If the request is bad, wouldn't it always be bad?

Reply:
It could be a temporary issue with the integration provider. Azure, Github and Gitlab all have bad days sometimes.

Reply:
@markstory And to clarify, we take the status code from the provider and use it in the response to the browser?

Reply:
No, getting a 400 requires the integration client code to catch `ApiError` and convert it into an `IntegrationError`. If the integration code doesn't handle `ApiError` then the endpoint will 500 which will still show user-feedback and be logged in sentry.